miio/device.py

```diff
@@ -155,6 +161,32 @@ def _fetch_info(self) -> DeviceInfo:
                 "Unable to request miIO.info from the device"
             ) from ex
 
+    def _initialize_descriptors(self) -> None:
+        """Cache all the descriptors once on the first call."""
+        if self._sensors is not None:
+            return
+
+        status = self.status()
+
+        # Sensors
+        self._sensors = status.sensors()
+
+        # Settings
+        self._settings = status.settings()
+        for setting in self._settings.values():
+            if setting.setter is None:
+                if setting.setter_name is None:
+                    raise Exception(
+                        f"Neither setter or setter_name was defined for {setting}"
+                    )
+                setting.setter = getattr(self, setting.setter_name)
+            if (
+                isinstance(setting, EnumSettingDescriptor)
```

Comment:
This should check if `setting.type == SettingType.Enum` instead of using the `isinstance`.

Reply:
thanks, done